tests/conftest.py

```diff
@@ -64,27 +76,6 @@ async def process_data(
     server.close()
 
 
-@pytest.fixture
```

Comment:
Maybe we can keep and replace `request_handler_protocol = protocol_factory()` with a protocol that does raw read/writes but that might be a bit complex to write a whole new one